core/src/test/java/org/apache/iceberg/util/TestParallelIterable.java

```diff
@@ -40,18 +40,41 @@ public class TestParallelIterable {
   public void closeParallelIteratorWithoutCompleteIteration()
       throws IOException, IllegalAccessException, NoSuchFieldException {
     ExecutorService executor = Executors.newFixedThreadPool(1);
+    Iterator<Integer> integerIterator =
+        new Iterator<Integer>() {
+          private int number = 1;
 
+          @Override
+          public boolean hasNext() {
+            if (number > 1000) {
+              return false;
+            }
+            number++;
+            return true;
+          }
+
+          @Override
+          public Integer next() {
+            try {
+              // sleep to control number generate rate
+              Thread.sleep(10);
+            } catch (InterruptedException e) {
+              System.out.println("Sleep interrupted, we ignore it!");
```

Comment:
we typically don't use sysout in tests, so better to remove it

Reply:
iceberg checkstyle don't support empty cache block, what do you suggest writing here?

Reply:
I add a comment

Reply:
you can add a checkstyle ignore via `@SuppressWarnings(...)` to the test method